src/lab13.py

```diff
@@ -760,15 +777,13 @@ def paint_visual_effects(node, cmds, rect):
     needs_blend_isolation = blend_mode != skia.BlendMode.kSrcOver or \
         needs_clip or opacity != 1.0
 
-    save_layer = \
-        SaveLayer(skia.Paint(BlendMode=blend_mode, Alphaf=opacity), node, [
-            ClipRRect(rect, clip_radius, cmds,
-                should_clip=needs_clip),
-        ], should_save=needs_blend_isolation)
-
-    transform = Transform(translation, rect, node, [save_layer])
-
-    node.save_layer = save_layer
+    blend_op = AlphaAndBlend(opacity, blend_mode, False, node, [
+        ClipRRect(rect, clip_radius,
```

Comment:
This needs to be AlphaOrBlend("destination-in", ...

Reply:
Can we delete ClipRRect also?

Reply:
Will do but I think separate PR, it is a little complicated due to the local absolute thing in a later chapter. (I think we want a ClipRRect operation even if it works by destination in blending.

Reply:
sg